### PR TITLE
Vision and autoref bridge

### LINKED ISSUE #15: Interface with SSL-Vision
It is required that we can take vision measurements from SSL-Vision and convert them into an internal type.

### PR DESCRIPTION
**Pull request by JNeiger**
Just wanted to throw this here to get some initial feedback about how I'm setting up the ROS packages. Still doesn't compile, but I can fix that later

- Moved protobufs to their own package so we can use them in the ref and vision package
- Moved multicast receiver into common package so we don't duplicate the code
- Started adding vision forwarding that matches the referee bridge style of implementation
- Added ref bridge

Closes #15 #13

### CODE REVIEW

**Comment by barulicm** [comment]
I like it. The packages you've split off make a lot of sense. 👍

**Comment by JNeiger** [comment]
Should be good to go. I'm seeing a few messages a second that aren't being parsed correctly, but all the data seems valid so far.

Like the autoref, I can't seem to get multicast working with docker grsim, only running locally. I'm using WSL so it may be that?

Also, this is diff'd onto `dev/barulicm/autoref_bridge`. We'll need to change it once that gets merged.

**Comment by JNeiger** [comment]
Per slack, we're just going to merge this in and have vision and referee stuff in the same PR. All is tested locally and works on my computer

File: .github/workflows/CI.yml
Comment (barulicm) [inline .github/workflows/CI.yml:21]: This can be installed by adding `protobuf-dev` as a dependency to the package that needs it. (This is the rosdep key that includes the `protobuf-compiler` apt package).

File: ateam_common/src/multicast_receiver.cpp
Comment (barulicm) [inline ateam_common/src/multicast_receiver.cpp:53]: Should we move this logging into the callback itself, and not need it to return anything? Could also make it easier to send this message to the ROS log.

File: ssl_league_msgs/msg/VisionDetectionRobot.msg
Comment (barulicm) [inline ssl_league_msgs/msg/VisionDetectionRobot.msg:3]: Thoughts on combining `pos` and `orientation` using `geometry_msgs/msg/Pose2D`? 
https://docs.ros2.org/galactic/api/geometry_msgs/msg/Pose2D.html

File: ssl_league_msgs/msg/VisionGeometryCameraCalibration.msg
Comment (barulicm) [inline ssl_league_msgs/msg/VisionGeometryCameraCalibration.msg:5]: Thoughts on combining `q` and `t` using `geometry_msgs/msg/Pose`?
https://docs.ros2.org/galactic/api/geometry_msgs/msg/Pose.html

**Review by barulicm — COMMENTED** [review]
Lookin good! Just a couple comments / questions.

File: ateam_common/src/multicast_receiver.cpp
Comment (JNeiger) [reply ateam_common/src/multicast_receiver.cpp]: That makes a lot of sense, done

File: ssl_league_msgs/msg/VisionDetectionRobot.msg
Comment (JNeiger) [reply ssl_league_msgs/msg/VisionDetectionRobot.msg]: Pose2D looks to be depreceated in Foxy, but I couldn't figure out a one liner to do yaw->quat so I'm cool just doing Pose2D until it finally dies and we swap.

Comment (barulicm) [reply ssl_league_msgs/msg/VisionDetectionRobot.msg]: Good catch, I completely missed that. 
One way to do yaw->quat is to use the utilities in `tf2_geometry_msgs`:
```
auto quat_msg = tf2::toMsg(tf2::Quaternion(tf2::Vector3(0,0,1), yaw));
```

File: ateam_autoref_bridge/src/autoref_bridge_node.cpp
Comment (barulicm) [inline ateam_autoref_bridge/src/autoref_bridge_node.cpp:24]: nit: should probably be `RCLCPP_WARN` or `RCLCPP_ERROR`.

**Comment by JNeiger** [comment]
Ayyy, got CI to pass, we good to go pending reviews

**Review by barulicm — APPROVED** [review]

**Pull request merged**
